src/main/java/com/jagrosh/jmusicbot/utils/TimeUtil.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2020 John Grosh <john.a.grosh@gmail.com>.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.jagrosh.jmusicbot.utils;
+
+public class TimeUtil
+{
+
+    public static String formatTime(long duration)
+    {
+        if(duration == Long.MAX_VALUE)
+            return "LIVE";
+        long seconds = Math.round(duration/1000.0);
+        long hours = seconds/(60*60);
+        seconds %= 60*60;
+        long minutes = seconds/60;
+        seconds %= 60;
+        return (hours>0 ? hours+":" : "") + (minutes<10 ? "0"+minutes : minutes) + ":" + (seconds<10 ? "0"+seconds : seconds);
+    }
+
+    /**
+     * @param args timestamp formatted as: [+ | -] &lt;HH:MM:SS | MM:SS | SS&gt;
+     * @return Time in milliseconds, negative if seeking backwards relatively
+     */
+    public static SeekTime parseTime(String args)
+    {
+        if (args.length() == 0 || args.length() > 8) return null;
+        String timestamp = args;
+        boolean relative = false; // seek forward or backward
+        boolean isSeekingBackwards = false;
+        char first = timestamp.charAt(0);
+        if (first == '+' || first == '-')
+        {
+            relative = true;
+            isSeekingBackwards = first == '-';
+            timestamp = timestamp.substring(1);
+        }
+
+        String[] timestampSplitArray = timestamp.split(":");
+        int unitTotal = timestampSplitArray.length;
+        if (unitTotal > 3) return null;
+
+        int[] timeUnitArray = new int[3]; // Hours, minutes, seconds
+        int timeUnitIndex = 3 - unitTotal;
+
+        for (String timeUnit : timestampSplitArray)
+        {
+            if (timeUnit.length() > 2 || timeUnit.length() == 0 || timeUnit.substring(0, 1).equals("+")) return null;
+            try
+            {
+                timeUnitArray[timeUnitIndex] = Integer.parseUnsignedInt(timeUnit);
+            } catch (NumberFormatException e)
+            {
+                return null;
+            }
+            timeUnitIndex++;
+        }
+
+        long milliseconds = timeUnitArray[0] * 3600000 + timeUnitArray[1] * 60000 + timeUnitArray[2] * 1000;
+        if (relative && isSeekingBackwards) milliseconds = -milliseconds;
```

Comment:
I think this can be simplified to:
```java
String[] timestampSplitArray = timestamp.split(":");
if(timestampSplitArray.length > 3 )
    return null;
int[] timeUnitArray = new int[3]; // hours, minutes, seconds
for(int index = 0; index < timestampSplitArray.length; index++)
{
    try
    {
        timeUnitArray[index + 3 - timestampSplitArray.length] = Integer.parseUnsignedInt(timestampSplitArray[index]);
    }
    catch (NumberFormatException e)
    {
        return null;
    }
}
long milliseconds = timeUnitArray[0] * 3600000 + timeUnitArray[1] * 60000 + timeUnitArray[2] * 1000;
milliseconds *= isSeekingBackwards ? -1 : 1;
```

Reply:
Yes, this looks better. Though I think it might be best to check for a positive sign (+) at the start of the time unit. According to an excerpt from the `Integer#parseUnsignedInt` doc: 
_The characters in the string must all be decimal digits, except that the first character may be an an ASCII plus sign '+' ('\u002B')._

Although it probably wouldn't hurt, being able to seek with `++01:00` seems weird to me.
Aaand I just noticed the doc has a typo. `may be an an ASCII`. Whew.